sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobServiceClientBuilder.java

```diff
@@ -52,13 +82,21 @@ public BlobServiceClient buildClient() {
      * @return a {@link BlobServiceAsyncClient} created from the configurations in this builder.
      */
     public BlobServiceAsyncClient buildAsyncClient() {
-        HttpPipeline pipeline = super.getPipeline();
-        if (pipeline == null) {
-            pipeline = super.buildPipeline();
-        }
+        HttpPipeline pipeline = (httpPipeline != null) ? httpPipeline : BuilderHelper.buildPipeline(() -> {
+            if (sharedKeyCredential != null) {
+                return new SharedKeyCredentialPolicy(sharedKeyCredential);
+            } else if (tokenCredential != null) {
+                return new BearerTokenAuthenticationPolicy(tokenCredential, String.format("%s/.default", endpoint));
```

Comment:
These special strings are used in multiple places, should there be some constants class?

Reply:
I think we can promote this later as it will need to go into public API.